packages/transaction-controller/src/types.ts

```diff
@@ -354,3 +360,20 @@ export interface DappSuggestedGasFees {
   maxFeePerGas?: string;
   maxPriorityFeePerGas?: string;
 }
+
+/**
+ * A transaction history operation that includes a note and timestamp.
+ */
+export type ExtendedHistoryOperation = Operation & {
+  note?: string;
+  timestamp?: number;
+};
+
+/**
+ * A transaction history that includes the transaction meta as the first element.
+ * And the rest of the elements are the operation arrays that were applied to the transaction meta.
+ */
+export type TransactionHistory = [
+  TransactionMeta,
+  ...ExtendedHistoryOperation[][],
```

Comment:
Would this be clearer if we also added a `TransactionHistoryEntry` type which was something like:

```
export type TransactionHistoryEntry = [
    ExtendedHistoryOperation,
    ...Operation[]
]
```

Then the `TransactionHistory` could just be:

```
[
   TransactionMeta,
   ...TransactionHistoryEntry[]
]
```

Reply:
Looks better, fixed in https://github.com/MetaMask/core/pull/1657/commits/c11e2708b26bb34211c27c9f8b815a9b2c78c569